lib/models/asyncapi.js

```diff
@@ -234,6 +234,121 @@ class AsyncAPIDocument extends Base {
   traverseSchemas(callback, schemaTypesToIterate) {
     traverseAsyncApiDocument(this, callback, schemaTypesToIterate);
   }
+
+  /**
+   * Converts a valid AsyncAPI document to a JavaScript Object Notation (JSON) string.
+   * A stringified AsyncAPI document using this function should be parsed via the AsyncAPIDocument.parse() function - the JSON.parse() function is not compatible.
+   * 
+   * @param {AsyncAPIDocument} doc A valid AsyncAPIDocument instance.
+   * @param {(number | string)=} space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
+   * @returns {string}
+   */
+  static stringify(doc, space) {
+    const rawDoc = doc.json();
+    const copiedDoc = { ...rawDoc };
+    copiedDoc[String(xParserSpecStringified)] = true;
+    return JSON.stringify(copiedDoc, refReplacer(), space);
+  }
+
+  /* eslint-disable sonarjs/cognitive-complexity */
```

Comment:
Instead of disabling sonar completely, would it not be an idea to have all this in a separate support file called `stringify.js` 🤔

Would make it much cleaner in terms of implementation as you can avoid nested functions. Also easier to test.

Reply:
I was thinking about it, but those functions are currently used only in AsyncAPIDocument class and I don't know if it makes sense to extract them. I also need nested functions to access variables defined above. I can always make them inline, but then I will have to pass those variables as arguments - is it worth it? 😄

Reply:
I mean, it just makes the functions complex to look at, as long as you tried to imagine how it would be for someone who has never seen the code, then all good 🙂 Your call.

Reply:
As inline they will still be complex 😆  Maybe a better solution would be to increase the complex function standard from 15 to 20 or 25? In many libraries we end up at around 17/18 and then this would fix the problem.

Reply:
The underlying problem is not sonar triggering cognitive-complexity 😄 You could easily refactor it to use smaller external functions and provide the missing variables as arguments, but it requires some restructuring of course.

Reply:
Fixed :P